lib/src/iterable_extensions.dart

```diff
@@ -166,6 +166,17 @@ extension IterableExtension<T> on Iterable<T> {
     }
   }
 
+  /// Returns an iterable containing only the non-`null` results
```

Comment:
Documentation should have a shorter first line.
I'd say:

```dart
/// The non-`null` results of calling [transform] on the elements of [this].
///
/// Returns a lazy iterable which calls [transform] 
/// on the elements of this iterable in iteration order,
/// then emits only the non-`null` values.
```
(Generally, I'd document iterables or collections by documenting their elements.)